apps/avatax/src/modules/public-log-drain/public-events.ts

```diff
@@ -0,0 +1,191 @@
+export const LogSeverityLevel = {
+  TRACE: "TRACE",
+  DEBUG: "DEBUG",
+  INFO: "INFO",
+  WARN: "WARN",
+  ERROR: "ERROR",
+  FATAL: "FATAL",
+} as const;
+
+export type LogSeverityLevelType = keyof typeof LogSeverityLevel;
+
+const EventTypes = {
+  TAXES_CALCULATED: "TAXES_CALCULATED",
+  TAXES_CALCULATION_FAILED: "TAXES_CALCULATION_FAILED",
+  AVATAX_TRANSACTION_CREATED: "AVATAX_TRANSACTION_CREATED",
+  AVATAX_TRANSACTION_CREATION_FAILED: "AVATAX_TRANSACTION_CREATION_FAILED",
+  SALEOR_ORDER_CONFIRMED: "SALEOR_ORDER_CONFIRMED",
+  SALEOR_ORDER_CONFIRMATION_FAILED: "SALEOR_ORDER_CONFIRMATION_FAILED",
+} as const;
+
+export interface PublicLog<T extends Record<string, unknown> = {}> {
+  message: string;
+
+  timestamp: Date;
+  level: LogSeverityLevelType;
+  attributes: T & {
+    saleorApiUrl: string;
+    eventType: keyof typeof EventTypes;
+  };
+}
+
+export class TaxesCalculatedInCheckoutLog implements PublicLog<{ checkoutId: string }> {
+  message = "Taxes calculated in checkout";
+  timestamp = new Date();
+  level = LogSeverityLevel.INFO;
+  attributes = { checkoutId: "", saleorApiUrl: "", eventType: EventTypes.TAXES_CALCULATED };
+
+  constructor(params: { checkoutId: string; saleorApiUrl: string }) {
+    this.attributes.checkoutId = params.checkoutId;
+    this.attributes.saleorApiUrl = params.saleorApiUrl;
+  }
+}
+
+export class TaxesCalculatedInOrderLog implements PublicLog<{ orderId: string }> {
+  message = "Taxes calculated in order";
+  timestamp = new Date();
+  level = LogSeverityLevel.INFO;
+  attributes = { orderId: "", saleorApiUrl: "", eventType: EventTypes.TAXES_CALCULATED };
+
+  constructor(params: { orderId: string; saleorApiUrl: string }) {
+    this.attributes.orderId = params.orderId;
+    this.attributes.saleorApiUrl = params.saleorApiUrl;
+  }
+}
+
+type CheckoutOrOrderId =
+  | {
+      orderId: string;
+      checkoutId?: undefined;
+    }
+  | {
+      orderId?: undefined;
+      checkoutId: string;
+    };
+
+class TaxesCalculationFailedLog implements PublicLog<CheckoutOrOrderId> {
+  message = "Taxes calculation failed";
```

Comment:
if we have separate events for successful calculations for checkout and order, we should probably be consistent here too